<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Appendix PD vs Controls, Padlocks</title>

<link href='../libs/fluidbox/css/fluidbox.css' rel='stylesheet' type='text/css'>
<link href='https://fonts.googleapis.com/css?family=Source+Code+Pro' rel='stylesheet' type='text/css'>
<link href='../include/css/style.css' rel='stylesheet' type='text/css'>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("section-TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#section-TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#section-TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#section-TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="section-TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <a class="navbar-brand" href="../IndexPage/">
        <img src="../include/gene.png" height="28" width="28">
      </a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="../IndexPage/">Home</a></li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Appendix<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Appendix_PDvsControls_Padlock/>PD vs Controls (padlock)</a></li><li><a href=../Appendix_PDvsControls_RNAseq/>PD vs Controls (RNA-seq)</a></li><li><a href=../Appendix_AgeAcceleration_Padlock/>Age*Diagnosis (padlock)</a></li><li><a href=../Appendix_AgeAcceleration_RNAseq/>Age*Diagnosis (RNAseq)</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Mice<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Mice_CecalPatch_Padlock/>Cecal Patch (padlock)</a></li><li><a href=../Mice_DSS_Padlock/>DSS (padlock)</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Brain<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Brain_PFC_Padlock_CGonly/>PFC II (padlock) CG only</a></li><li><a href=../Brain_PFCRep_Padlock_withGLU/>PFC (large) + GLU (padlock)</a></li><li><a href=../Brain_PFCRep_Padlock_withGLU_Braak/>PFC (large) + GLU Stage-wise (padlock)</a></li><li><a href=../Brain_OFB_Padlock_CGonly/>OFB (padlock) CG only</a></li><li><a href=../Brain_AgeAcceleration_Padlock/>Age*Diagnosis in PFC (large) (padlock)</a></li><li><a href=../Brain_PFC_hMeDIPseq/>PFC (hMeDIP)</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Discover<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Discover_Appendix_Meth2Expression/>Appendix Methylation vs Expression</a></li><li><a href=../Discover_OverlapAppendixMice/>Overlap Appendix & Mice</a></li><li><a href=../Discover_OverlapAppendixBrain/>Overlap Appendix & Brain</a></li><li><a href=../Discover_GenomicEnrichment/>Genomic enrichment</a></li><li><a href=../Discover_Aging/>Aging and Disease Interaction</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Figures<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Figure1/>Figure 1</a></li><li><a href=../Figure2/>Figure 2</a></li><li><a href=../Figure3/>Figure 3</a></li><li><a href=../Figure4/>Figure 4</a></li><li><a href=../Figure5/>Figure 5</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Results<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Results_Appendix/>Appendix</a></li><li><a href=../Results_Brain/>Brain</a></li><li><a href=../Results_Proteomics/>Proteomics</a></li><li><a href=../Results_Aging/>Aging</a></li><li><a href=../Results_Mice/>Mice</a></li><li><a href=../Results_etc/>etc</a></li>
          </ul>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="section-header">



<h1 class="title toc-ignore">Appendix PD vs Controls, Padlocks</h1>

</div>


<!-- # (c)  Juozas Gordevi<c4><8d>ius -->
<div id="section-parameters" class="section level1">
<h1>Parameters</h1>
<pre><code>## {
##   &quot;host&quot;: &quot;DL560&quot;,
##   &quot;nPerm&quot;: 40,
##   &quot;workers&quot;: 41,
##   &quot;outliers&quot;: {
##     &quot;thresholdSD&quot;: 3,
##     &quot;noOfPCs&quot;: 2,
##     &quot;removeOutliers&quot;: true,
##     &quot;shape&quot;: &quot;Diagnosis&quot;
##   },
##   &quot;clustering&quot;: {
##     &quot;groups&quot;: [&quot;Diagnosis&quot;, &quot;Batch&quot;, &quot;Sex&quot;, &quot;substr(ID, 1, 1)&quot;],
##     &quot;addClustersToModel&quot;: 0
##   },
##   &quot;normalization&quot;: {
##     &quot;normalize&quot;: true,
##     &quot;plotColor&quot;: [&quot;Diagnosis&quot;, &quot;Batch&quot;]
##   },
##   &quot;imputation&quot;: {
##     &quot;imputeBetas&quot;: false,
##     &quot;imputeCounts&quot;: false
##   },
##   &quot;model&quot;: {
##     &quot;formula&quot;: &quot;~ Diagnosis + .&quot;,
##     &quot;formula0&quot;: &quot;~ . - Diagnosis&quot;,
##     &quot;svRange&quot;: [1, 2, 3, 4, 5, 6]
##   }
## }</code></pre>
</div>
<div id="section-data" class="section level1">
<h1>Data</h1>
<p>From directory PD_Appendix_padloc_01172019, the beta and count matrices provided by Peipei. Technical replicates were combined by averaging and rounding their M and U counts. Beta values were already averaged.</p>
<p>Resulting data structure:</p>
<pre><code>##      sample   beta genome counts countsU countsM modelData
## [1,]     44 186080 186080 186080  186080  186080        44
## [2,]      6     44      6     44      44      44         5</code></pre>
<!-- Run the processing pipeline -->
</div>
<div id="section-analysis" class="section level1">
<h1>Analysis</h1>
<div id="section-outlier-identification" class="section level2">
<h2>Outlier identification</h2>
<p><img src="index_files/figure-html/outliers_plotPCAPairs-1.png" width="100%" /></p>
<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:right;">
Age
</th>
<th style="text-align:left;">
Sex
</th>
<th style="text-align:right;">
PMI
</th>
<th style="text-align:left;">
Diagnosis
</th>
<th style="text-align:left;">
Batch
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
X3616
</td>
<td style="text-align:right;">
87
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:left;">
PD/LBD
</td>
<td style="text-align:left;">
2
</td>
</tr>
</tbody>
</table>
<blockquote>
<p>Outliers have been removed.</p>
</blockquote>
<p>[1] “untouched tables: genome”</p>
<!-- 
  @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
  Clustering before normalization 
-->
</div>
<div id="section-sample-clustering" class="section level2">
<h2>Sample clustering</h2>
<p>Hierarchical clustering using correlations as similarity measure and Ward linkage.</p>
<p><img src="index_files/figure-html/unnamed-chunk-9-1.png" width="100%" /></p>
<!-- 
  @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
  Normalization
-->
</div>
<div id="section-normalization" class="section level2">
<h2>Normalization</h2>
<p>Signal density before normalization</p>
<p><img src="index_files/figure-html/unnamed-chunk-12-1.png" width="100%" /></p>
<!-- After normalization -->
<p>NU and M matrices were quantile normalized separately, total counts and beta values were obtained from normalized matrices.</p>
<p>Signals after normalization</p>
<p><img src="index_files/figure-html/unnamed-chunk-13-1.png" width="100%" />PCA and outliers after normalization</p>
<p><img src="index_files/figure-html/unnamed-chunk-13-2.png" width="100%" /></p>
</div>
<div id="section-covariates" class="section level2">
<h2>Covariates</h2>
<p>Covariates used in subsequent models</p>
<pre><code>## [1] &quot;Diagnosis&quot; &quot;Age&quot;       &quot;Gender&quot;    &quot;PMI&quot;       &quot;Cohort&quot;</code></pre>
<p>Are any of the covariates related to the variable of interest? (Assuming that the first variable in modelData is the one of interest).</p>
<pre><code>## 
## Call:
## glm(formula = formula, family = &quot;binomial&quot;, data = data$modelData)
## 
## Deviance Residuals: 
##     Min       1Q   Median       3Q      Max  
## -2.3085  -0.8423   0.5606   0.8336   1.7668  
## 
## Coefficients:
##             Estimate Std. Error z value Pr(&gt;|z|)  
## (Intercept) -7.44857    3.10950  -2.395   0.0166 *
## Age          0.09126    0.03906   2.336   0.0195 *
## GenderM      0.24555    0.73720   0.333   0.7391  
## PMI          0.01940    0.03181   0.610   0.5419  
## Cohort2      1.31004    0.90441   1.449   0.1475  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for binomial family taken to be 1)
## 
##     Null deviance: 59.028  on 42  degrees of freedom
## Residual deviance: 48.507  on 38  degrees of freedom
## AIC: 58.507
## 
## Number of Fisher Scoring iterations: 3</code></pre>
</div>
</div>
<div id="section-robust-lmfit" class="section level1">
<h1>Robust lmFit</h1>
<p>50% stable, non extreme change, M values, robust limma eBayes</p>
<pre class="r"><code>methylation &lt;- data$beta
prefix &lt;- &quot;m6_&quot;
modelparams &lt;- list(
    allowStable = 0.5,
    useMvalues  = TRUE,
    fixInfinity = 0.001, # Fraction of change for 0% and 100% methylated locations
    downloadDataMatrix = TRUE
  )</code></pre>
<div id="section-removal-of-non-informative-loci" class="section level2">
<h2>Removal of non-informative loci</h2>
<p>Fraction of CGs that should be removed depending on threshold. Currently set threshold is 50%</p>
<blockquote>
<p>Removing 4056 stable loci.</p>
</blockquote>
</div>
<div id="section-conversion-of-beta-values-to-m-values" class="section level2">
<h2>Conversion of Beta values to M values</h2>
<blockquote>
<p>Beta values were converted to M values.</p>
</blockquote>
<p><img src="index_files/figure-html/unnamed-chunk-20-1.png" width="100%" /></p>
</div>
<div id="section-data-download" class="section level2">
<h2>Data download</h2>
<blockquote>
<p>Download <a href="www/m6_methylation.RDS">data</a> file.</p>
</blockquote>
</div>
<div id="section-data-pcs-vs-covariates" class="section level2">
<h2>Data PCs vs Covariates</h2>
<p>Fit covariates to the PCs using limma</p>
<pre><code>## Removing intercept from test coefficients</code></pre>
<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
DiagnosisPD.LBD
</th>
<th style="text-align:right;">
Age
</th>
<th style="text-align:right;">
GenderM
</th>
<th style="text-align:right;">
PMI
</th>
<th style="text-align:right;">
Cohort2
</th>
<th style="text-align:right;">
AveExpr
</th>
<th style="text-align:right;">
F
</th>
<th style="text-align:right;">
P.Value
</th>
<th style="text-align:right;">
adj.P.Val
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
PC2
</td>
<td style="text-align:right;">
76.904703
</td>
<td style="text-align:right;">
-1.0547219
</td>
<td style="text-align:right;">
-14.7864775
</td>
<td style="text-align:right;">
0.3016495
</td>
<td style="text-align:right;">
146.362970
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
11.1925726
</td>
<td style="text-align:right;">
0.0000002
</td>
<td style="text-align:right;">
0.0000016
</td>
</tr>
<tr>
<td style="text-align:left;">
PC1
</td>
<td style="text-align:right;">
22.248383
</td>
<td style="text-align:right;">
2.5607461
</td>
<td style="text-align:right;">
-5.2079937
</td>
<td style="text-align:right;">
3.1536798
</td>
<td style="text-align:right;">
-162.325266
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
4.7266008
</td>
<td style="text-align:right;">
0.0011179
</td>
<td style="text-align:right;">
0.0055894
</td>
</tr>
<tr>
<td style="text-align:left;">
PC10
</td>
<td style="text-align:right;">
14.276310
</td>
<td style="text-align:right;">
0.2534325
</td>
<td style="text-align:right;">
-52.6082515
</td>
<td style="text-align:right;">
0.1433894
</td>
<td style="text-align:right;">
-33.344847
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.4622049
</td>
<td style="text-align:right;">
0.2165833
</td>
<td style="text-align:right;">
0.7219443
</td>
</tr>
<tr>
<td style="text-align:left;">
PC7
</td>
<td style="text-align:right;">
3.450729
</td>
<td style="text-align:right;">
-0.5503676
</td>
<td style="text-align:right;">
49.9101706
</td>
<td style="text-align:right;">
-0.8809700
</td>
<td style="text-align:right;">
-18.141146
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.2078040
</td>
<td style="text-align:right;">
0.3173068
</td>
<td style="text-align:right;">
0.7932670
</td>
</tr>
<tr>
<td style="text-align:left;">
PC6
</td>
<td style="text-align:right;">
6.619927
</td>
<td style="text-align:right;">
-1.4137807
</td>
<td style="text-align:right;">
-21.7775736
</td>
<td style="text-align:right;">
0.8743630
</td>
<td style="text-align:right;">
29.060463
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.6822517
</td>
<td style="text-align:right;">
0.6387353
</td>
<td style="text-align:right;">
0.9854582
</td>
</tr>
<tr>
<td style="text-align:left;">
PC4
</td>
<td style="text-align:right;">
1.843375
</td>
<td style="text-align:right;">
-1.2829620
</td>
<td style="text-align:right;">
-0.1692827
</td>
<td style="text-align:right;">
1.0403035
</td>
<td style="text-align:right;">
1.294628
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.3870650
</td>
<td style="text-align:right;">
0.8556474
</td>
<td style="text-align:right;">
0.9854582
</td>
</tr>
<tr>
<td style="text-align:left;">
PC3
</td>
<td style="text-align:right;">
13.016268
</td>
<td style="text-align:right;">
-0.8858720
</td>
<td style="text-align:right;">
-1.7703018
</td>
<td style="text-align:right;">
0.2443311
</td>
<td style="text-align:right;">
-33.507155
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.3449713
</td>
<td style="text-align:right;">
0.8833717
</td>
<td style="text-align:right;">
0.9854582
</td>
</tr>
<tr>
<td style="text-align:left;">
PC9
</td>
<td style="text-align:right;">
3.339050
</td>
<td style="text-align:right;">
0.8389194
</td>
<td style="text-align:right;">
18.7402740
</td>
<td style="text-align:right;">
-0.3801317
</td>
<td style="text-align:right;">
-9.241741
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.2939885
</td>
<td style="text-align:right;">
0.9143208
</td>
<td style="text-align:right;">
0.9854582
</td>
</tr>
<tr>
<td style="text-align:left;">
PC8
</td>
<td style="text-align:right;">
-4.409340
</td>
<td style="text-align:right;">
0.0554741
</td>
<td style="text-align:right;">
-14.0074651
</td>
<td style="text-align:right;">
-0.6086494
</td>
<td style="text-align:right;">
-4.742636
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.1515609
</td>
<td style="text-align:right;">
0.9787557
</td>
<td style="text-align:right;">
0.9854582
</td>
</tr>
<tr>
<td style="text-align:left;">
PC5
</td>
<td style="text-align:right;">
-1.775371
</td>
<td style="text-align:right;">
-0.5252749
</td>
<td style="text-align:right;">
11.8084775
</td>
<td style="text-align:right;">
0.1424838
</td>
<td style="text-align:right;">
-9.170617
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.1279488
</td>
<td style="text-align:right;">
0.9854582
</td>
<td style="text-align:right;">
0.9854582
</td>
</tr>
</tbody>
</table>
<p>The covariates significantly affecting the PCs using ANOVA:</p>
<p><img src="index_files/figure-html/unnamed-chunk-24-1.png" width="100%" /></p>
</div>
<div id="section-select-the-number-of-svs" class="section level2">
<h2>Select the number of SVs</h2>
<p>All known covariates are regressed from data and PCA of residuals is computed. The number of PCs that explain more than 5% of variability is used as the desired number of SVs.</p>
<pre><code>## Selected number of SVs 1</code></pre>
</div>
<div id="section-probe-level-models" class="section level2">
<h2>Probe-level models</h2>
<div id="section-p-value-distribution" class="section level3">
<h3>P value distribution</h3>
<pre><code>## Warning: Removed 29 rows containing non-finite values (stat_bin).</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-29-1.png" width="100%" /></p>
</div>
<div id="section-volcano-plot" class="section level3">
<h3>Volcano plot</h3>
<p><img src="index_files/figure-html/unnamed-chunk-31-1.png" width="100%" /></p>
</div>
<div id="section-manhattan-plot" class="section level3">
<h3>Manhattan plot</h3>
<p><img src="index_files/figure-html/unnamed-chunk-32-1.png" width="100%" /><a href="www/m6_manhattan.pdf">PDF</a></p>
</div>
<div id="section-enrichment" class="section level3">
<h3>Enrichment</h3>
<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
substr
</th>
<th style="text-align:right;">
MeanFC
</th>
<th style="text-align:right;">
NoSign
</th>
<th style="text-align:right;">
OR
</th>
<th style="text-align:right;">
P
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
CG
</td>
<td style="text-align:right;">
-0.0152209
</td>
<td style="text-align:right;">
928
</td>
<td style="text-align:right;">
1.3864
</td>
<td style="text-align:right;">
1e-06
</td>
</tr>
</tbody>
</table>
</div>
<div id="section-download" class="section level3">
<h3>Download</h3>
<p><a href="www/m6_Appendix_PDvsControls_Padlock.csv">Download results</a></p>
<!-- Final touches to remove lasting effects -->
</div>
</div>
</div>
<div id="section-sessioninfo" class="section level1">
<h1>SessionInfo</h1>
<pre><code>## R version 3.5.1 (2018-07-02)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /opt/intel/compilers_and_libraries_2018.2.199/linux/mkl/lib/intel64_lin/libmkl_rt.so
## 
## locale:
##  [1] LC_CTYPE=C                 LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] doSNOW_1.0.16         snow_0.4-3            itertools_0.1-3      
##  [4] iterators_1.0.10      limma_3.38.3          impute_1.56.0        
##  [7] WGCNA_1.66            fastcluster_1.1.25    dynamicTreeCut_1.63-1
## [10] glue_1.3.0            foreach_1.4.4         ggpubr_0.2           
## [13] magrittr_1.5          plotly_4.8.0          ggplot2_3.1.0        
## [16] dplyr_0.7.8           data.table_1.12.0     kableExtra_1.0.1     
## [19] knitr_1.21            pacman_0.5.0          shiny_1.2.0          
## 
## loaded via a namespace (and not attached):
##  [1] matrixStats_0.54.0    fit.models_0.5-14     robust_0.4-18        
##  [4] bit64_0.9-7           doParallel_1.0.14     webshot_0.5.1        
##  [7] RColorBrewer_1.1-2    httr_1.4.0            tools_3.5.1          
## [10] backports_1.1.3       R6_2.3.0              rpart_4.1-13         
## [13] Hmisc_4.2-0           DBI_1.0.0             lazyeval_0.2.1       
## [16] BiocGenerics_0.28.0   colorspace_1.4-0      nnet_7.3-12          
## [19] withr_2.1.2           tidyselect_0.2.5      gridExtra_2.3        
## [22] preprocessCore_1.44.0 bit_1.1-14            compiler_3.5.1       
## [25] rvest_0.3.2           Biobase_2.42.0        htmlTable_1.13.1     
## [28] xml2_1.2.0            labeling_0.3          scales_1.0.0         
## [31] checkmate_1.9.1       mvtnorm_1.0-8         DEoptimR_1.0-8       
## [34] robustbase_0.93-3     readr_1.3.1           stringr_1.3.1        
## [37] digest_0.6.18         foreign_0.8-71        rmarkdown_1.11       
## [40] rrcov_1.4-7           base64enc_0.1-3       pkgconfig_2.0.2      
## [43] htmltools_0.3.6       highr_0.7             htmlwidgets_1.3      
## [46] rlang_0.3.1           rstudioapi_0.9.0      RSQLite_2.1.1        
## [49] bindr_0.1.1           jsonlite_1.6          acepack_1.4.1        
## [52] GO.db_3.7.0           Formula_1.2-3         Matrix_1.2-15        
## [55] Rcpp_1.0.0            munsell_0.5.0         S4Vectors_0.20.1     
## [58] stringi_1.2.4         yaml_2.2.0            MASS_7.3-51.1        
## [61] plyr_1.8.4            grid_3.5.1            blob_1.1.1           
## [64] ggrepel_0.8.0         promises_1.0.1        crayon_1.3.4         
## [67] lattice_0.20-38       cowplot_0.9.4         splines_3.5.1        
## [70] hms_0.4.2             pillar_1.3.1          codetools_0.2-16     
## [73] stats4_3.5.1          evaluate_0.12         latticeExtra_0.6-28  
## [76] httpuv_1.4.5.1        gtable_0.2.0          purrr_0.3.0          
## [79] tidyr_0.8.2           assertthat_0.2.0      xfun_0.4             
## [82] mime_0.6              xtable_1.8-3          later_0.7.5          
## [85] pcaPP_1.9-73          survival_2.43-3       viridisLite_0.3.0    
## [88] tibble_2.0.1          AnnotationDbi_1.44.0  memoise_1.1.0        
## [91] IRanges_2.16.0        bindrcpp_0.2.2        cluster_2.0.7-1</code></pre>
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="dependencies">
{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["bootstrap"]},{"type":"character","attributes":{},"value":["3.3.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/bootstrap"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["viewport"]}},"value":[{"type":"character","attributes":{},"value":["width=device-width, initial-scale=1"]}]},{"type":"character","attributes":{},"value":["js/bootstrap.min.js","shim/html5shiv.min.js","shim/respond.min.js"]},{"type":"character","attributes":{},"value":["css/bootstrap.min.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jqueryui"]},{"type":"character","attributes":{},"value":["1.11.4"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jqueryui"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery-ui.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tocify"]},{"type":"character","attributes":{},"value":["1.9.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/tocify"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.tocify.js"]},{"type":"character","attributes":{},"value":["jquery.tocify.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["navigation"]},{"type":"character","attributes":{},"value":["1.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/navigation-1.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tabsets.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["highlightjs"]},{"type":"character","attributes":{},"value":["9.12.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/highlightjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["highlight.js"]},{"type":"character","attributes":{},"value":["default.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["kePrint"]},{"type":"character","attributes":{},"value":["0.0.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["kePrint-0.0.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["kePrint.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["kableExtra"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.0.1"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["kePrint"]},{"type":"character","attributes":{},"value":["0.0.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["kePrint-0.0.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["kePrint.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["kableExtra"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.0.1"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["kePrint"]},{"type":"character","attributes":{},"value":["0.0.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["kePrint-0.0.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["kePrint.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["kableExtra"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.0.1"]}]}]}
</script>
<!--/html_preserve-->
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="execution_dependencies">
{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages","version"]},"class":{"type":"character","attributes":{},"value":["data.frame"]},"row.names":{"type":"integer","attributes":{},"value":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122]}},"value":[{"type":"character","attributes":{},"value":["acepack","AnnotationDbi","assertthat","backports","base","base64enc","bindr","bindrcpp","Biobase","BiocGenerics","bit","bit64","blob","checkmate","cluster","codetools","colorspace","compiler","cowplot","crayon","data.table","datasets","DBI","DEoptimR","digest","doParallel","doSNOW","dplyr","dynamicTreeCut","evaluate","fastcluster","fit.models","foreach","foreign","Formula","ggplot2","ggpubr","ggrepel","glue","GO.db","graphics","grDevices","grid","gridExtra","gtable","highr","Hmisc","hms","htmlTable","htmltools","htmlwidgets","httpuv","httr","impute","IRanges","iterators","itertools","jsonlite","kableExtra","knitr","labeling","later","lattice","latticeExtra","lazyeval","limma","magrittr","MASS","Matrix","matrixStats","memoise","methods","mime","munsell","mvtnorm","nnet","pacman","parallel","pcaPP","pillar","pkgconfig","plotly","plyr","preprocessCore","promises","purrr","R6","RColorBrewer","Rcpp","readr","rlang","rmarkdown","robust","robustbase","rpart","rrcov","RSQLite","rstudioapi","rvest","S4Vectors","scales","shiny","snow","splines","stats","stats4","stringi","stringr","survival","tibble","tidyr","tidyselect","tools","utils","viridisLite","webshot","WGCNA","withr","xfun","xml2","xtable","yaml"]},{"type":"character","attributes":{},"value":["1.4.1","1.44.0","0.2.0","1.1.3","3.5.1","0.1-3","0.1.1","0.2.2","2.42.0","0.28.0","1.1-14","0.9-7","1.1.1","1.9.1","2.0.7-1","0.2-16","1.4-0","3.5.1","0.9.4","1.3.4","1.12.0","3.5.1","1.0.0","1.0-8","0.6.18","1.0.14","1.0.16","0.7.8","1.63-1","0.12","1.1.25","0.5-14","1.4.4","0.8-71","1.2-3","3.1.0","0.2","0.8.0","1.3.0","3.7.0","3.5.1","3.5.1","3.5.1","2.3","0.2.0","0.7","4.2-0","0.4.2","1.13.1","0.3.6","1.3","1.4.5.1","1.4.0","1.56.0","2.16.0","1.0.10","0.1-3","1.6","1.0.1","1.21","0.3","0.7.5","0.20-38","0.6-28","0.2.1","3.38.3","1.5","7.3-51.1","1.2-15","0.54.0","1.1.0","3.5.1","0.6","0.5.0","1.0-8","7.3-12","0.5.0","3.5.1","1.9-73","1.3.1","2.0.2","4.8.0","1.8.4","1.44.0","1.0.1","0.3.0","2.3.0","1.1-2","1.0.0","1.3.1","0.3.1","1.11","0.4-18","0.93-3","4.1-13","1.4-7","2.1.1","0.9.0","0.3.2","0.20.1","1.0.0","1.2.0","0.4-3","3.5.1","3.5.1","3.5.1","1.2.4","1.3.1","2.43-3","2.0.1","0.8.2","0.2.5","3.5.1","3.5.1","0.3.0","0.5.1","1.66","2.1.2","0.4","1.2.0","1.8-3","2.2.0"]}]}]}
</script>
<!--/html_preserve-->
</div>

<script type="text/javascript" src="../libs/fluidbox/jquery.fluidbox.min.js"></script>
<script type="text/javascript" src="../include/js/enable-fluidbox.js"></script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
