<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Methylation 2 Expression, Appendix</title>

<link href='../libs/fluidbox/css/fluidbox.css' rel='stylesheet' type='text/css'>
<link href='https://fonts.googleapis.com/css?family=Source+Code+Pro' rel='stylesheet' type='text/css'>
<link href='../include/css/style.css' rel='stylesheet' type='text/css'>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("section-TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#section-TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#section-TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#section-TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="section-TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <a class="navbar-brand" href="../IndexPage/">
        <img src="../include/gene.png" height="28" width="28">
      </a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="../IndexPage/">Home</a></li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Appendix<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Appendix_PDvsControls_Padlock/>PD vs Controls (padlock)</a></li><li><a href=../Appendix_PDvsControls_RNAseq/>PD vs Controls (RNA-seq)</a></li><li><a href=../Appendix_AgeAcceleration_Padlock/>Age*Diagnosis (padlock)</a></li><li><a href=../Appendix_AgeAcceleration_RNAseq/>Age*Diagnosis (RNAseq)</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Mice<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Mice_CecalPatch_Padlock/>Cecal Patch (padlock)</a></li><li><a href=../Mice_DSS_Padlock/>DSS (padlock)</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Brain<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Brain_PFC_Padlock_CGonly/>PFC II (padlock) CG only</a></li><li><a href=../Brain_PFCRep_Padlock_withGLU/>PFC (large) + GLU (padlock)</a></li><li><a href=../Brain_PFCRep_Padlock_withGLU_Braak/>PFC (large) + GLU Stage-wise (padlock)</a></li><li><a href=../Brain_OFB_Padlock_CGonly/>OFB (padlock) CG only</a></li><li><a href=../Brain_AgeAcceleration_Padlock/>Age*Diagnosis in PFC (large) (padlock)</a></li><li><a href=../Brain_PFC_hMeDIPseq/>PFC (hMeDIP)</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Discover<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Discover_Appendix_Meth2Expression/>Appendix Methylation vs Expression</a></li><li><a href=../Discover_OverlapAppendixMice/>Overlap Appendix & Mice</a></li><li><a href=../Discover_OverlapAppendixBrain/>Overlap Appendix & Brain</a></li><li><a href=../Discover_GenomicEnrichment/>Genomic enrichment</a></li><li><a href=../Discover_Aging/>Aging and Disease Interaction</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Figures<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Figure1/>Figure 1</a></li><li><a href=../Figure2/>Figure 2</a></li><li><a href=../Figure3/>Figure 3</a></li><li><a href=../Figure4/>Figure 4</a></li><li><a href=../Figure5/>Figure 5</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Results<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Results_Appendix/>Appendix</a></li><li><a href=../Results_Brain/>Brain</a></li><li><a href=../Results_Proteomics/>Proteomics</a></li><li><a href=../Results_Aging/>Aging</a></li><li><a href=../Results_Mice/>Mice</a></li><li><a href=../Results_etc/>etc</a></li>
          </ul>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="section-header">



<h1 class="title toc-ignore">Methylation 2 Expression, Appendix</h1>

</div>


<!-- # (c)  Juozas Gordevi<c4><8d>ius -->
<div id="section-correlation-between-methylation-and-expression" class="section level1">
<h1>Correlation between methylation and expression</h1>
<p>Find loci that are in gene promoters. Average methylation signal in promoters. Match samples across meth and RNA. See if gene expression correlates with methylation.</p>
<p>Number of significant correlations by sign</p>
<pre><code>## 
## -1  1 
## 13  3</code></pre>
<p>Volcano plot of correlations</p>
<p><img src="index_files/figure-html/unnamed-chunk-5-1.png" width="100%" /></p>
</div>
<div id="section-enrichment-of-autophagy-pathways-in-rna-seq" class="section level1">
<h1>Enrichment of autophagy pathways in RNA-seq</h1>
<p>Enrichment of AP target genes among DEGs (q &lt; 0.2) in RNA-seq</p>
<pre><code>##        InAP
## Sig     FALSE  TRUE
##   FALSE 12951   380
##   TRUE    959    46</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.003623
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  1.168365 2.241648
## sample estimates:
## odds ratio 
##   1.634701</code></pre>
<p>Are any of the autophagy pathways enriched with significant genes or have higher FC or lower p value?</p>
<p>Selective autophagy</p>
<pre><code>## [[1]]
## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.1456
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  0.0792216 1.2444756
## sample estimates:
## odds ratio 
##  0.3989095 
## 
## 
## [[2]]
## 
##  Welch Two Sample t-test
## 
## data:  V1 by V2
## t = 4.0521, df = 48.029, p-value = 0.0001847
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.02269129 0.06738775
## sample estimates:
## mean in group FALSE  mean in group TRUE 
##          0.13244020          0.08740068 
## 
## 
## [[3]]
## 
##  Welch Two Sample t-test
## 
## data:  V1 by V2
## t = 2.7273, df = 48.012, p-value = 0.008892
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.09869924 0.65251643
## sample estimates:
## mean in group FALSE  mean in group TRUE 
##            1.534005            1.158397</code></pre>
<p>Macroautophagy</p>
<pre><code>## [[1]]
## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.8607
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  0.6514216 1.3550844
## sample estimates:
## odds ratio 
##  0.9513644 
## 
## 
## [[2]]
## 
##  Welch Two Sample t-test
## 
## data:  V1 by V2
## t = 5.8074, df = 294.87, p-value = 1.641e-08
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.02114982 0.04283247
## sample estimates:
## mean in group FALSE  mean in group TRUE 
##           0.1328919           0.1009008 
## 
## 
## [[3]]
## 
##  Welch Two Sample t-test
## 
## data:  V1 by V2
## t = 0.826, df = 281.88, p-value = 0.4095
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.1074285  0.2627777
## sample estimates:
## mean in group FALSE  mean in group TRUE 
##            1.534210            1.456536</code></pre>
<p>Lysosomes</p>
<pre><code>## Batch submitting query [==========================] 100% eta: 0s</code></pre>
<pre><code>## [[1]]
## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 1
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  0.7703206 1.2714600
## sample estimates:
## odds ratio 
##  0.9950403 
## 
## 
## [[2]]
## 
##  Welch Two Sample t-test
## 
## data:  V1 by V2
## t = 0.17192, df = 609.06, p-value = 0.8636
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.009800882  0.011681480
## sample estimates:
## mean in group FALSE  mean in group TRUE 
##           0.1323258           0.1313855 
## 
## 
## [[3]]
## 
##  Welch Two Sample t-test
## 
## data:  V1 by V2
## t = -1.3478, df = 588.13, p-value = 0.1782
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.27649888  0.05144589
## sample estimates:
## mean in group FALSE  mean in group TRUE 
##            1.528391            1.640917</code></pre>
<p>Chaperone-mediated autophagy</p>
<pre><code>## [[1]]
## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.1504
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  0.000000 1.671413
## sample estimates:
## odds ratio 
##          0 
## 
## 
## [[2]]
## 
##  Welch Two Sample t-test
## 
## data:  V1 by V2
## t = 1.9849, df = 14.064, p-value = 0.06702
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.003824596  0.099325638
## sample estimates:
## mean in group FALSE  mean in group TRUE 
##          0.13233936          0.08458884 
## 
## 
## [[3]]
## 
##  Welch Two Sample t-test
## 
## data:  V1 by V2
## t = 5.9333, df = 14.225, p-value = 3.419e-05
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.6024484 1.2829992
## sample estimates:
## mean in group FALSE  mean in group TRUE 
##           1.5337336           0.5910098</code></pre>
<p>Autophagy</p>
<pre><code>## Batch submitting query [==========================] 100% eta: 0s</code></pre>
<pre><code>## [[1]]
## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.6799
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  0.6934872 1.2290161
## sample estimates:
## odds ratio 
##    0.93005 
## 
## 
## [[2]]
## 
##  Welch Two Sample t-test
## 
## data:  V1 by V2
## t = 5.1486, df = 503.55, p-value = 3.77e-07
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.01530179 0.03418623
## sample estimates:
## mean in group FALSE  mean in group TRUE 
##           0.1330592           0.1083152 
## 
## 
## [[3]]
## 
##  Welch Two Sample t-test
## 
## data:  V1 by V2
## t = 0.93287, df = 478.97, p-value = 0.3514
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.07816924  0.21948157
## sample estimates:
## mean in group FALSE  mean in group TRUE 
##            1.534945            1.464289</code></pre>
</div>
<div id="section-overlap-of-targeted-dmc-genes-and-degs" class="section level1">
<h1>Overlap of targeted DMC genes and DEGs</h1>
</div>
<div id="section-overlap-of-genes-with-dmc-in-genic-region-and-rna-seq" class="section level1">
<h1>Overlap of genes with DMC in genic region and rna-seq</h1>
<p>Enrichment of autophagy genes with significant DMCs in their promoters</p>
<pre class="r"><code>autophagyList &lt;- meth$fit[, unique(Gene)]
annot[grepl(&quot;hg19_genes_[1to5kb|promoters]&quot;, annot.type), sum(adj.P.Val &lt; 0.05), annot.symbol] %&gt;%
  .[, table(annot.symbol %in% autophagyList, V1 &gt; 0)] %&gt;%
  tee %&gt;% fisher.test</code></pre>
<pre><code>##        
##         FALSE TRUE
##   FALSE  2872  336
##   TRUE    337   52</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.08385
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  0.9447527 1.8130551
## sample estimates:
## odds ratio 
##   1.318809</code></pre>
<pre class="r"><code>annot[grepl(&quot;hg19_genes_[1to5kb|promoters]&quot;, annot.type), sum(adj.P.Val &lt; 0.05 &amp; logFC &gt; 0), annot.symbol] %&gt;%
  .[, table(annot.symbol %in% autophagyList, V1 &gt; 0)] %&gt;%
  tee %&gt;% fisher.test</code></pre>
<pre><code>##        
##         FALSE TRUE
##   FALSE  2984  224
##   TRUE    346   43</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.00555
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  1.143764 2.351301
## sample estimates:
## odds ratio 
##   1.655276</code></pre>
<pre class="r"><code>annot[grepl(&quot;hg19_genes_[1to5kb|promoters]&quot;, annot.type), sum(adj.P.Val &lt; 0.05 &amp; logFC &lt; 0), annot.symbol] %&gt;%
  .[, table(annot.symbol %in% autophagyList, V1 &gt; 0)] %&gt;%
  tee %&gt;% fisher.test</code></pre>
<pre><code>##        
##         FALSE TRUE
##   FALSE  3087  121
##   TRUE    379   10</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.3139
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  0.3120828 1.2960026
## sample estimates:
## odds ratio 
##   0.673215</code></pre>
<p>Find the padlock target genes that have DMCs in promoters. Are they more likely to be DE than other genes?</p>
<pre class="r"><code>geneList &lt;- annot[
  grepl(&quot;hg19_genes_[1to5kb|promoters]&quot;, annot.type) &amp; 
    adj.P.Val &lt; 0.05 &amp;
    annot.symbol %in% meth$fit[, unique(Gene)],
  unique(annot.symbol)
]
length(geneList)</code></pre>
<pre><code>## [1] 52</code></pre>
<pre class="r"><code>rna$fit[, table(SYMBOL %in% geneList, adj.P.Val &lt; 0.2)] %&gt;%
  tee %&gt;%
  fisher.test</code></pre>
<pre><code>##        
##         FALSE  TRUE
##   FALSE 13286  1002
##   TRUE     45     3</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 1
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  0.1754082 2.7615510
## sample estimates:
## odds ratio 
##  0.8839686</code></pre>
<pre class="r"><code>rna$fit[SYMBOL %in% meth$fit[, unique(Gene)], table(SYMBOL %in% geneList, adj.P.Val &lt; 0.2)] %&gt;%
  tee %&gt;%
  fisher.test</code></pre>
<pre><code>##        
##         FALSE TRUE
##   FALSE   335   43
##   TRUE     45    3</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.4562
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  0.09914696 1.73376644
## sample estimates:
## odds ratio 
##  0.5200386</code></pre>
<p>Are the DE genes with DMC in their promoter more likely to belong to any of the autophagy processes?</p>
<p>Selective autophagy</p>
<pre><code>##        
##         FALSE TRUE
##   FALSE   400   15
##   TRUE     11    0</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 1
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##   0.00000 11.79343
## sample estimates:
## odds ratio 
##          0</code></pre>
<p>Macroautophagy</p>
<pre><code>##        
##         FALSE TRUE
##   FALSE   337   78
##   TRUE      9    2</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 1
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  0.09906792 4.77100842
## sample estimates:
## odds ratio 
##  0.9601927</code></pre>
<p>Lysosomes</p>
<pre><code>##        
##         FALSE TRUE
##   FALSE   189  226
##   TRUE      5    6</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 1
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  0.2507917 4.2259806
## sample estimates:
## odds ratio 
##   1.003532</code></pre>
<p>Chaperone-mediated autophagy</p>
<pre><code>##        
##         FALSE TRUE
##   FALSE   408    7
##   TRUE     11    0</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 1
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##   0.00000 29.11259
## sample estimates:
## odds ratio 
##          0</code></pre>
<p>Autophagy</p>
<pre><code>##        
##         FALSE TRUE
##   FALSE   305  110
##   TRUE      7    4</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.4948
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  0.3331663 6.3672319
## sample estimates:
## odds ratio 
##   1.582543</code></pre>
</div>
<div id="section-fig-1-d" class="section level1">
<h1>Fig 1 D</h1>
<p>Show how many genes in a pathway X have DMCs within 300kb (padlock target)</p>
<p>How many genes in a pathway X have DMCs within their promoter?</p>
</div>
<div id="section-fig-1-e" class="section level1">
<h1>Fig 1 E</h1>
<p>Gene expression volcano with DE &amp; AP genes labeled</p>
</div>
<div id="section-enrichment-of-upregulation-within-pathways" class="section level1">
<h1>Enrichment of upregulation within pathways</h1>
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="dependencies">
{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["bootstrap"]},{"type":"character","attributes":{},"value":["3.3.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/bootstrap"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["viewport"]}},"value":[{"type":"character","attributes":{},"value":["width=device-width, initial-scale=1"]}]},{"type":"character","attributes":{},"value":["js/bootstrap.min.js","shim/html5shiv.min.js","shim/respond.min.js"]},{"type":"character","attributes":{},"value":["css/bootstrap.min.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jqueryui"]},{"type":"character","attributes":{},"value":["1.11.4"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jqueryui"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery-ui.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tocify"]},{"type":"character","attributes":{},"value":["1.9.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/tocify"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.tocify.js"]},{"type":"character","attributes":{},"value":["jquery.tocify.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["navigation"]},{"type":"character","attributes":{},"value":["1.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/navigation-1.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tabsets.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["highlightjs"]},{"type":"character","attributes":{},"value":["9.12.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/highlightjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["highlight.js"]},{"type":"character","attributes":{},"value":["default.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]}]}
</script>
<!--/html_preserve-->
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="execution_dependencies">
{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages","version"]},"class":{"type":"character","attributes":{},"value":["data.frame"]},"row.names":{"type":"integer","attributes":{},"value":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152]}},"value":[{"type":"character","attributes":{},"value":["acepack","AnnotationDbi","AnnotationHub","annotatr","assertthat","backports","base","base64enc","bindr","bindrcpp","Biobase","BiocGenerics","BiocManager","BiocParallel","biomaRt","Biostrings","bit","bit64","bitops","blob","broom","BSgenome","checkmate","cluster","codetools","colorspace","compiler","crayon","curl","data.table","datasets","DBI","DelayedArray","DEoptimR","digest","doParallel","doSNOW","dplyr","dynamicTreeCut","evaluate","fastcluster","fit.models","foreach","foreign","Formula","generics","GenomeInfoDb","GenomeInfoDbData","GenomicAlignments","GenomicFeatures","GenomicRanges","ggplot2","ggpubr","ggrepel","glue","GO.db","graphics","grDevices","grid","gridExtra","gtable","Hmisc","hms","htmlTable","htmltools","htmlwidgets","httpuv","httr","impute","interactiveDisplayBase","IRanges","iterators","itertools","jsonlite","kableExtra","knitr","labeling","later","lattice","latticeExtra","lazyeval","limma","magrittr","MASS","Matrix","matrixStats","memoise","methods","mime","munsell","mvtnorm","nlme","nnet","org.Hs.eg.db","pacman","parallel","pcaPP","pillar","pkgconfig","plotly","plyr","preprocessCore","prettyunits","progress","promises","purrr","R6","RColorBrewer","Rcpp","RCurl","readr","regioneR","reshape2","rlang","rmarkdown","robust","robustbase","rpart","rrcov","Rsamtools","RSQLite","rstudioapi","rtracklayer","rvest","S4Vectors","scales","shiny","snow","splines","stats","stats4","stringi","stringr","SummarizedExperiment","survival","tibble","tidyr","tidyselect","tools","TxDb.Hsapiens.UCSC.hg19.knownGene","utils","viridisLite","webshot","WGCNA","withr","xfun","XML","xml2","xtable","XVector","yaml","zlibbioc"]},{"type":"character","attributes":{},"value":["1.4.1","1.44.0","2.14.5","1.8.0","0.2.0","1.1.3","3.5.1","0.1-3","0.1.1","0.2.2","2.42.0","0.28.0","1.30.4","1.16.5","2.38.0","2.50.2","1.1-14","0.9-7","1.0-6","1.1.1","0.5.1","1.50.0","1.9.1","2.0.7-1","0.2-16","1.4-0","3.5.1","1.3.4","3.3","1.12.0","3.5.1","1.0.0","0.8.0","1.0-8","0.6.18","1.0.14","1.0.16","0.7.8","1.63-1","0.12","1.1.25","0.5-14","1.4.4","0.8-71","1.2-3","0.0.2","1.18.1","1.2.0","1.18.1","1.34.3","1.34.0","3.1.0","0.2","0.8.0","1.3.0","3.7.0","3.5.1","3.5.1","3.5.1","2.3","0.2.0","4.2-0","0.4.2","1.13.1","0.3.6","1.3","1.4.5.1","1.4.0","1.56.0","1.20.0","2.16.0","1.0.10","0.1-3","1.6","1.0.1","1.21","0.3","0.7.5","0.20-38","0.6-28","0.2.1","3.38.3","1.5","7.3-51.1","1.2-15","0.54.0","1.1.0","3.5.1","0.6","0.5.0","1.0-8","3.1-137","7.3-12","3.7.0","0.5.0","3.5.1","1.9-73","1.3.1","2.0.2","4.8.0","1.8.4","1.44.0","1.0.2","1.2.0","1.0.1","0.3.0","2.3.0","1.1-2","1.0.0","1.95-4.11","1.3.1","1.14.0","1.4.3","0.3.1","1.11","0.4-18","0.93-3","4.1-13","1.4-7","1.34.0","2.1.1","0.9.0","1.42.1","0.3.2","0.20.1","1.0.0","1.2.0","0.4-3","3.5.1","3.5.1","3.5.1","1.2.4","1.3.1","1.12.0","2.43-3","2.0.1","0.8.2","0.2.5","3.5.1","3.2.2","3.5.1","0.3.0","0.5.1","1.66","2.1.2","0.4","3.98-1.16","1.2.0","1.8-3","0.22.0","2.2.0","1.28.0"]}]}]}
</script>
<!--/html_preserve-->
</div>

<script type="text/javascript" src="../libs/fluidbox/jquery.fluidbox.min.js"></script>
<script type="text/javascript" src="../include/js/enable-fluidbox.js"></script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
