---
title: "etc"
output:
  html_document:
    toc: true
    toc_depth: 2
    toc_float: true
    includes:
      in_header: ../include/in_header.html
      before_body: ../include/before_body.html
      after_body: ../include/after_body.html
runtime: shiny_prerendered
---
<!-- # (c)  Juozas Gordevičius -->

```{r setup, include=FALSE}
source("../code/common.R")
```

```{r}
# Data primers
if (!dir.exists("www")) 
	dir.create('www')

file.copy("../etc/Supplementary Data_primers.xlsx", "www/")
```

```{r}

```

# Supplementary tables


```{r}
url <- "./www/Supplementary Data_primers.xlsx"
stopifnot(file.exists(url))
```

- [Supp. table 16](`r url`) Padlock primers


```{r}
url <- "https://docs.google.com/spreadsheets/d/14Z7YIxQUEkHXsMOHdDYaVeiHVDX4dOulHMfC5mq2MoU/edit?usp=sharing"
```

- [Supp. table 17](`r url`) Experimental parameters



```{r}
studies <- list.files(path = "../", pattern = "*Padlock*|*RNAseq*")

res <-
foreach (study = studies) %do% {
	message("Loading study: ", study)
	path <- file.path("..", study, "www")
	name <- paste0("m6_methylation.RDS")
	if (file.exists(file.path(path, name))) {
		# padlock study
		x <- readRDS(file.path(path, name))
		x$key[, Study := study]
	} else {
		name <- name <- paste0("r1_rnaseq.RDS")
		if (file.exists(file.path(path, name))) {
			# rnaseq study
			x <- readRDS(file.path(path, name))
			x$key[, Study := study]
		} else {
			message(study, ": no file?")
		}
	}
}
names(res) <- studies

p_load("xlsx")
file <- "www/sampleKey.xlsx"
unlink(file)
for (study in studies) {
	write.xlsx(res[[study]], file, sheetName=study, 
	  col.names=TRUE, row.names=FALSE, append=TRUE)
}

url <- "./www/sampleKey.xlsx"
```

- [Supp. table 18](`r url`) Sample description for each study, xlsx file












