<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Appendix Age Acceleration, Padlocks</title>

<link href='../libs/fluidbox/css/fluidbox.css' rel='stylesheet' type='text/css'>
<link href='https://fonts.googleapis.com/css?family=Source+Code+Pro' rel='stylesheet' type='text/css'>
<link href='../include/css/style.css' rel='stylesheet' type='text/css'>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("section-TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#section-TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#section-TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#section-TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="section-TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <a class="navbar-brand" href="../IndexPage/">
        <img src="../include/gene.png" height="28" width="28">
      </a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="../IndexPage/">Home</a></li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Appendix<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Appendix_PDvsControls_Padlock/>PD vs Controls (padlock)</a></li><li><a href=../Appendix_PDvsControls_RNAseq/>PD vs Controls (RNA-seq)</a></li><li><a href=../Appendix_AgeAcceleration_Padlock/>Age*Diagnosis (padlock)</a></li><li><a href=../Appendix_AgeAcceleration_RNAseq/>Age*Diagnosis (RNAseq)</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Mice<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Mice_CecalPatch_Padlock/>Cecal Patch (padlock)</a></li><li><a href=../Mice_DSS_Padlock/>DSS (padlock)</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Brain<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Brain_PFC_Padlock_CGonly/>PFC II (padlock) CG only</a></li><li><a href=../Brain_PFCRep_Padlock_withGLU/>PFC (large) + GLU (padlock)</a></li><li><a href=../Brain_PFCRep_Padlock_withGLU_Braak/>PFC (large) + GLU Stage-wise (padlock)</a></li><li><a href=../Brain_OFB_Padlock_CGonly/>OFB (padlock) CG only</a></li><li><a href=../Brain_AgeAcceleration_Padlock/>Age*Diagnosis in PFC (large) (padlock)</a></li><li><a href=../Brain_PFC_hMeDIPseq/>PFC (hMeDIP)</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Discover<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Discover_Appendix_Meth2Expression/>Appendix Methylation vs Expression</a></li><li><a href=../Discover_OverlapAppendixMice/>Overlap Appendix & Mice</a></li><li><a href=../Discover_OverlapAppendixBrain/>Overlap Appendix & Brain</a></li><li><a href=../Discover_GenomicEnrichment/>Genomic enrichment</a></li><li><a href=../Discover_Aging/>Aging and Disease Interaction</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Figures<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Figure1/>Figure 1</a></li><li><a href=../Figure2/>Figure 2</a></li><li><a href=../Figure3/>Figure 3</a></li><li><a href=../Figure4/>Figure 4</a></li><li><a href=../Figure5/>Figure 5</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Results<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Results_Appendix/>Appendix</a></li><li><a href=../Results_Brain/>Brain</a></li><li><a href=../Results_Proteomics/>Proteomics</a></li><li><a href=../Results_Aging/>Aging</a></li><li><a href=../Results_Mice/>Mice</a></li><li><a href=../Results_etc/>etc</a></li>
          </ul>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="section-header">



<h1 class="title toc-ignore">Appendix Age Acceleration, Padlocks</h1>

</div>


<!-- # (c)  Juozas Gordevi<c4><8d>ius -->
<div id="section-parameters" class="section level1">
<h1>Parameters</h1>
<pre><code>## {
##   &quot;host&quot;: &quot;DL560&quot;,
##   &quot;nPerm&quot;: 40,
##   &quot;workers&quot;: 41,
##   &quot;outliers&quot;: {
##     &quot;thresholdSD&quot;: 3,
##     &quot;noOfPCs&quot;: 2,
##     &quot;removeOutliers&quot;: true,
##     &quot;shape&quot;: &quot;Diagnosis&quot;
##   },
##   &quot;clustering&quot;: {
##     &quot;groups&quot;: [&quot;Sex&quot;, &quot;Batch&quot;, &quot;substr(ID, 1, 1)&quot;],
##     &quot;addClustersToModel&quot;: 0
##   },
##   &quot;normalization&quot;: {
##     &quot;normalize&quot;: true,
##     &quot;plotColor&quot;: &quot;Batch&quot;
##   },
##   &quot;imputation&quot;: {
##     &quot;imputeBetas&quot;: false,
##     &quot;imputeCounts&quot;: false
##   },
##   &quot;model&quot;: {
##     &quot;formula&quot;: &quot;~ 0 + Diagnosis + Diagnosis:Age + Cohort + Sex + PMI&quot;,
##     &quot;formula0&quot;: &quot;~ 0 + Diagnosis + Age + Cohort + Sex + PMI&quot;,
##     &quot;contrasts&quot;: [&quot;DiagnosisControl.Age&quot;, &quot;DiagnosisPD.LBD.Age&quot;, &quot;DiagnosisControl.Age - DiagnosisPD.LBD.Age&quot;],
##     &quot;svRange&quot;: [1, 2, 3, 4, 5, 6]
##   }
## }</code></pre>
</div>
<div id="section-data" class="section level1">
<h1>Data</h1>
<p>Combined together all appendix samples</p>
<p>Resulting data structure:</p>
<pre><code>##        beta counts countsU countsM genome sample modelData
## [1,] 185400 185400  185400  185400 185400     77        77
## [2,]     77     77      77      77      6      6         5</code></pre>
<!-- Run the processing pipeline -->
</div>
<div id="section-analysis" class="section level1">
<h1>Analysis</h1>
<div id="section-outlier-identification" class="section level2">
<h2>Outlier identification</h2>
<p><img src="index_files/figure-html/outliers_plotPCAPairs-1.png" width="100%" /></p>
<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Diagnosis
</th>
<th style="text-align:right;">
Age
</th>
<th style="text-align:left;">
Sex
</th>
<th style="text-align:right;">
PMI
</th>
<th style="text-align:left;">
Batch
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
E1
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:right;">
77
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:left;">
0
</td>
</tr>
</tbody>
</table>
<blockquote>
<p>Outliers have been removed.</p>
</blockquote>
<p>[1] “untouched tables: genome”</p>
<!-- 
  @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
  Clustering before normalization 
-->
</div>
<div id="section-sample-clustering" class="section level2">
<h2>Sample clustering</h2>
<p>Hierarchical clustering using correlations as similarity measure and Ward linkage.</p>
<p><img src="index_files/figure-html/unnamed-chunk-9-1.png" width="100%" /></p>
<!-- 
  @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
  Normalization
-->
</div>
<div id="section-normalization" class="section level2">
<h2>Normalization</h2>
<p>Signal density before normalization</p>
<p><img src="index_files/figure-html/unnamed-chunk-12-1.png" width="100%" /></p>
<!-- After normalization -->
<p>NU and M matrices were quantile normalized separately, total counts and beta values were obtained from normalized matrices.</p>
<p>Signals after normalization</p>
<p><img src="index_files/figure-html/unnamed-chunk-13-1.png" width="100%" />PCA and outliers after normalization</p>
<p><img src="index_files/figure-html/unnamed-chunk-13-2.png" width="100%" /></p>
</div>
<div id="section-covariates" class="section level2">
<h2>Covariates</h2>
<p>Covariates used in subsequent models</p>
<pre><code>## [1] &quot;Age&quot;       &quot;Sex&quot;       &quot;PMI&quot;       &quot;Diagnosis&quot; &quot;Cohort&quot;</code></pre>
<p>Are any of the covariates related to the variable of interest? (Assuming that the first variable in modelData is the one of interest).</p>
<pre><code>## 
## Call:
## glm(formula = formula, data = data$modelData)
## 
## Deviance Residuals: 
##     Min       1Q   Median       3Q      Max  
## -37.530   -7.877    0.092    8.597   27.459  
## 
## Coefficients:
##                 Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)     61.68481    5.59199  11.031   &lt;2e-16 ***
## SexF            -3.10099    3.52384  -0.880   0.3819    
## PMI             -0.06786    0.11148  -0.609   0.5447    
## DiagnosisPD/LBD  8.14520    4.70745   1.730   0.0880 .  
## Cohort1         11.65611    5.06100   2.303   0.0242 *  
## Cohort2         13.34032    7.17953   1.858   0.0674 .  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for gaussian family taken to be 217.9247)
## 
##     Null deviance: 22772  on 75  degrees of freedom
## Residual deviance: 15255  on 70  degrees of freedom
## AIC: 632.62
## 
## Number of Fisher Scoring iterations: 2</code></pre>
</div>
</div>
<div id="section-robust-lmfit-v2" class="section level1">
<h1>Robust lmFit v2</h1>
<p>50% stable, non extreme change, M values, robust limma eBayes</p>
<pre class="r"><code>methylation &lt;- data$beta
prefix &lt;- &quot;m6_&quot;
modelparams &lt;- list(
    allowStable = 0.5,
    useMvalues  = TRUE,
    fixInfinity = 0.001, # Fraction of change for 0% and 100% methylated locations
    downloadDataMatrix = TRUE,
    accelerationVariable = &quot;Diagnosis&quot; # Name of variable to test differential acceleration
  )</code></pre>
<div id="section-removal-of-non-informative-loci" class="section level2">
<h2>Removal of non-informative loci</h2>
<p>Fraction of CGs that should be removed depending on threshold. Currently set threshold is 50%</p>
<blockquote>
<p>Removing 4149 stable loci.</p>
</blockquote>
</div>
<div id="section-conversion-of-beta-values-to-m-values" class="section level2">
<h2>Conversion of Beta values to M values</h2>
<blockquote>
<p>Beta values were converted to M values.</p>
</blockquote>
<p><img src="index_files/figure-html/unnamed-chunk-20-1.png" width="100%" /></p>
</div>
<div id="section-data-download" class="section level2">
<h2>Data download</h2>
<blockquote>
<p>Download <a href="www/m6_methylation.RDS">data</a> file.</p>
</blockquote>
</div>
<div id="section-data-pcs-vs-covariates" class="section level2">
<h2>Data PCs vs Covariates</h2>
<p>Fit covariates to the PCs using limma</p>
<pre><code>## Removing intercept from test coefficients</code></pre>
<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Age
</th>
<th style="text-align:right;">
SexF
</th>
<th style="text-align:right;">
PMI
</th>
<th style="text-align:right;">
DiagnosisPD.LBD
</th>
<th style="text-align:right;">
Cohort1
</th>
<th style="text-align:right;">
Cohort2
</th>
<th style="text-align:right;">
AveExpr
</th>
<th style="text-align:right;">
F
</th>
<th style="text-align:right;">
P.Value
</th>
<th style="text-align:right;">
adj.P.Val
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
PC2
</td>
<td style="text-align:right;">
0.0509270
</td>
<td style="text-align:right;">
-3.949484
</td>
<td style="text-align:right;">
0.3931122
</td>
<td style="text-align:right;">
-50.2154148
</td>
<td style="text-align:right;">
27.3658647
</td>
<td style="text-align:right;">
-166.267664
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
36.4544308
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
PC1
</td>
<td style="text-align:right;">
1.4606977
</td>
<td style="text-align:right;">
-9.747071
</td>
<td style="text-align:right;">
0.9693784
</td>
<td style="text-align:right;">
46.8310065
</td>
<td style="text-align:right;">
134.2670783
</td>
<td style="text-align:right;">
17.817949
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
6.2807021
</td>
<td style="text-align:right;">
0.0000207
</td>
<td style="text-align:right;">
0.0001037
</td>
</tr>
<tr>
<td style="text-align:left;">
PC3
</td>
<td style="text-align:right;">
1.0965194
</td>
<td style="text-align:right;">
-5.872748
</td>
<td style="text-align:right;">
-1.2094009
</td>
<td style="text-align:right;">
-13.2816173
</td>
<td style="text-align:right;">
-19.5589566
</td>
<td style="text-align:right;">
-44.330161
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.9350130
</td>
<td style="text-align:right;">
0.0856121
</td>
<td style="text-align:right;">
0.2853737
</td>
</tr>
<tr>
<td style="text-align:left;">
PC10
</td>
<td style="text-align:right;">
-0.5602210
</td>
<td style="text-align:right;">
-15.965172
</td>
<td style="text-align:right;">
0.1423908
</td>
<td style="text-align:right;">
0.9733939
</td>
<td style="text-align:right;">
16.5172242
</td>
<td style="text-align:right;">
49.303571
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.1835284
</td>
<td style="text-align:right;">
0.3239432
</td>
<td style="text-align:right;">
0.8098581
</td>
</tr>
<tr>
<td style="text-align:left;">
PC7
</td>
<td style="text-align:right;">
0.0001914
</td>
<td style="text-align:right;">
-20.596684
</td>
<td style="text-align:right;">
-0.0434452
</td>
<td style="text-align:right;">
4.4237900
</td>
<td style="text-align:right;">
-10.8329012
</td>
<td style="text-align:right;">
-41.326421
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.9349055
</td>
<td style="text-align:right;">
0.4750344
</td>
<td style="text-align:right;">
0.8841988
</td>
</tr>
<tr>
<td style="text-align:left;">
PC8
</td>
<td style="text-align:right;">
0.4126289
</td>
<td style="text-align:right;">
25.878483
</td>
<td style="text-align:right;">
0.3981055
</td>
<td style="text-align:right;">
-5.1136487
</td>
<td style="text-align:right;">
7.1813230
</td>
<td style="text-align:right;">
12.086213
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.7951254
</td>
<td style="text-align:right;">
0.5765839
</td>
<td style="text-align:right;">
0.8841988
</td>
</tr>
<tr>
<td style="text-align:left;">
PC4
</td>
<td style="text-align:right;">
-0.5064738
</td>
<td style="text-align:right;">
12.084177
</td>
<td style="text-align:right;">
0.6432703
</td>
<td style="text-align:right;">
13.8977172
</td>
<td style="text-align:right;">
0.3506204
</td>
<td style="text-align:right;">
5.775859
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.7401322
</td>
<td style="text-align:right;">
0.6189392
</td>
<td style="text-align:right;">
0.8841988
</td>
</tr>
<tr>
<td style="text-align:left;">
PC6
</td>
<td style="text-align:right;">
-0.4317315
</td>
<td style="text-align:right;">
12.937660
</td>
<td style="text-align:right;">
0.2684138
</td>
<td style="text-align:right;">
5.1614074
</td>
<td style="text-align:right;">
4.0233476
</td>
<td style="text-align:right;">
26.428917
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.5558536
</td>
<td style="text-align:right;">
0.7639839
</td>
<td style="text-align:right;">
0.9549799
</td>
</tr>
<tr>
<td style="text-align:left;">
PC9
</td>
<td style="text-align:right;">
0.0424355
</td>
<td style="text-align:right;">
-12.632568
</td>
<td style="text-align:right;">
0.0523024
</td>
<td style="text-align:right;">
-5.8565731
</td>
<td style="text-align:right;">
15.2715485
</td>
<td style="text-align:right;">
17.618872
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.3644055
</td>
<td style="text-align:right;">
0.8992779
</td>
<td style="text-align:right;">
0.9933597
</td>
</tr>
<tr>
<td style="text-align:left;">
PC5
</td>
<td style="text-align:right;">
0.2504043
</td>
<td style="text-align:right;">
9.115773
</td>
<td style="text-align:right;">
0.0210117
</td>
<td style="text-align:right;">
0.2902436
</td>
<td style="text-align:right;">
-5.3102160
</td>
<td style="text-align:right;">
1.053021
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.1223718
</td>
<td style="text-align:right;">
0.9933597
</td>
<td style="text-align:right;">
0.9933597
</td>
</tr>
</tbody>
</table>
<p>The covariates significantly affecting the PCs using ANOVA:</p>
<p><img src="index_files/figure-html/unnamed-chunk-24-1.png" width="100%" /></p>
</div>
<div id="section-select-the-number-of-svs" class="section level2">
<h2>Select the number of SVs</h2>
<p>All known covariates are regressed from data and PCA of residuals is computed. The number of PCs that explain more than 5% of variability is used as the desired number of SVs.</p>
<pre><code>## Selected number of SVs 0</code></pre>
</div>
<div id="section-probe-level-models" class="section level2">
<h2>Probe-level models</h2>
<div id="section-p-value-distribution" class="section level3">
<h3>P value distribution</h3>
<pre><code>## Warning: Removed 30 rows containing non-finite values (stat_bin).</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-29-1.png" width="100%" /></p>
</div>
<div id="section-volcano-plot" class="section level3">
<h3>Volcano plot</h3>
<p><img src="index_files/figure-html/unnamed-chunk-31-1.png" width="100%" /></p>
</div>
<div id="section-manhattan-plot" class="section level3">
<h3>Manhattan plot</h3>
<p><img src="index_files/figure-html/unnamed-chunk-32-1.png" width="100%" /><a href="www/m6_manhattan_DiagnosisControl.Age.pdf">PDF</a> <img src="index_files/figure-html/unnamed-chunk-32-2.png" width="100%" /><a href="www/m6_manhattan_DiagnosisPD.LBD.Age.pdf">PDF</a> <img src="index_files/figure-html/unnamed-chunk-32-3.png" width="100%" /><a href="www/m6_manhattan_DiagnosisControl.Age-DiagnosisPD.LBD.Age.pdf">PDF</a></p>
</div>
<div id="section-enrichment" class="section level3">
<h3>Enrichment</h3>
<pre><code>## NULL</code></pre>
</div>
<div id="section-download" class="section level3">
<h3>Download</h3>
<p><a href="www/m6_Appendix_AgeAcceleration_Padlock.csv">Download results</a></p>
<!-- Final touches to remove lasting effects -->
</div>
<div id="section-age-acceleration" class="section level3">
<h3>Age acceleration</h3>
<p>An age predictor is trained and evaluated using 10 fold cross-validation on all the samples. Random forest is used as training algorithm.</p>
<p><img src="index_files/figure-html/unnamed-chunk-39-1.png" width="100%" /></p>
</div>
</div>
</div>
<div id="section-sessioninfo" class="section level1">
<h1>SessionInfo</h1>
<pre><code>## R version 3.5.1 (2018-07-02)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /opt/intel/compilers_and_libraries_2018.2.199/linux/mkl/lib/intel64_lin/libmkl_rt.so
## 
## locale:
##  [1] LC_CTYPE=C                 LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] bindrcpp_0.2.2        ggsci_2.9             randomForest_4.6-14  
##  [4] caret_6.0-81          lattice_0.20-38       doSNOW_1.0.16        
##  [7] snow_0.4-3            itertools_0.1-3       iterators_1.0.10     
## [10] limma_3.38.3          impute_1.56.0         WGCNA_1.66           
## [13] fastcluster_1.1.25    dynamicTreeCut_1.63-1 glue_1.3.0           
## [16] foreach_1.4.4         ggpubr_0.2            magrittr_1.5         
## [19] plotly_4.8.0          ggplot2_3.1.0         dplyr_0.7.8          
## [22] data.table_1.12.0     kableExtra_1.0.1      knitr_1.21           
## [25] pacman_0.5.0          shiny_1.2.0          
## 
## loaded via a namespace (and not attached):
##   [1] colorspace_1.4-0      class_7.3-15          htmlTable_1.13.1     
##   [4] base64enc_0.1-3       rstudioapi_0.9.0      ggrepel_0.8.0        
##   [7] bit64_0.9-7           lubridate_1.7.4       prodlim_2018.04.18   
##  [10] AnnotationDbi_1.44.0  mvtnorm_1.0-8         xml2_1.2.0           
##  [13] codetools_0.2-16      splines_3.5.1         doParallel_1.0.14    
##  [16] robustbase_0.93-3     Formula_1.2-3         jsonlite_1.6         
##  [19] cluster_2.0.7-1       GO.db_3.7.0           rrcov_1.4-7          
##  [22] readr_1.3.1           compiler_3.5.1        httr_1.4.0           
##  [25] backports_1.1.3       assertthat_0.2.0      Matrix_1.2-15        
##  [28] lazyeval_0.2.1        later_0.7.5           acepack_1.4.1        
##  [31] htmltools_0.3.6       tools_3.5.1           gtable_0.2.0         
##  [34] reshape2_1.4.3        Rcpp_1.0.0            Biobase_2.42.0       
##  [37] nlme_3.1-137          preprocessCore_1.44.0 timeDate_3043.102    
##  [40] gower_0.1.2           xfun_0.4              stringr_1.3.1        
##  [43] rvest_0.3.2           mime_0.6              DEoptimR_1.0-8       
##  [46] MASS_7.3-51.1         scales_1.0.0          ipred_0.9-8          
##  [49] hms_0.4.2             promises_1.0.1        RColorBrewer_1.1-2   
##  [52] yaml_2.2.0            memoise_1.1.0         gridExtra_2.3        
##  [55] rpart_4.1-13          latticeExtra_0.6-28   stringi_1.2.4        
##  [58] RSQLite_2.1.1         highr_0.7             S4Vectors_0.20.1     
##  [61] pcaPP_1.9-73          checkmate_1.9.1       BiocGenerics_0.28.0  
##  [64] lava_1.6.5            rlang_0.3.1           pkgconfig_2.0.2      
##  [67] matrixStats_0.54.0    evaluate_0.12         purrr_0.3.0          
##  [70] bindr_0.1.1           recipes_0.1.4         htmlwidgets_1.3      
##  [73] labeling_0.3          cowplot_0.9.4         bit_1.1-14           
##  [76] tidyselect_0.2.5      robust_0.4-18         plyr_1.8.4           
##  [79] R6_2.3.0              generics_0.0.2        IRanges_2.16.0       
##  [82] Hmisc_4.2-0           fit.models_0.5-14     DBI_1.0.0            
##  [85] pillar_1.3.1          foreign_0.8-71        withr_2.1.2          
##  [88] survival_2.43-3       nnet_7.3-12           tibble_2.0.1         
##  [91] crayon_1.3.4          rmarkdown_1.11        grid_3.5.1           
##  [94] blob_1.1.1            ModelMetrics_1.2.2    digest_0.6.18        
##  [97] webshot_0.5.1         xtable_1.8-3          tidyr_0.8.2          
## [100] httpuv_1.4.5.1        stats4_3.5.1          munsell_0.5.0        
## [103] viridisLite_0.3.0</code></pre>

<script type="application/shiny-prerendered" data-context="server-start">
source("../code/common.R")
source("../code/plots.R")
source("../code/linearFits.R")
source("../code/manhattanPlot.R")

params <- list(
  host          = system("hostname", intern=TRUE),
  nPerm         = 40,
  workers       = 41,
  outliers      = list(
    thresholdSD = 3,
    noOfPCs     = 2,
    removeOutliers = TRUE,
    shape       = "Diagnosis"
  ),
  clustering    = list(
    groups      = c(Gender="Sex", Cohort="Batch", 
      Plate="substr(ID, 1, 1)"),
    addClustersToModel = 0
  ),
  normalization = list(
    normalize   = TRUE,
    plotColor   = c("Batch")
  ),
  imputation    = list(
    imputeBetas = FALSE,
    imputeCounts= FALSE
  ),
  model         = list(
    formula     = "~ 0 + Diagnosis + Diagnosis:Age + Cohort + Sex + PMI",
    formula0    = "~ 0 + Diagnosis + Age + Cohort + Sex + PMI",
    contrasts = c("DiagnosisControl.Age", "DiagnosisPD.LBD.Age", "DiagnosisControl.Age - DiagnosisPD.LBD.Age"),
    svRange     = 1:6
  )
)

</script>
 <!--html_preserve-->
<script type="application/shiny-prerendered" data-context="dependencies">
{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["bootstrap"]},{"type":"character","attributes":{},"value":["3.3.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/bootstrap"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["viewport"]}},"value":[{"type":"character","attributes":{},"value":["width=device-width, initial-scale=1"]}]},{"type":"character","attributes":{},"value":["js/bootstrap.min.js","shim/html5shiv.min.js","shim/respond.min.js"]},{"type":"character","attributes":{},"value":["css/bootstrap.min.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jqueryui"]},{"type":"character","attributes":{},"value":["1.11.4"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jqueryui"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery-ui.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tocify"]},{"type":"character","attributes":{},"value":["1.9.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/tocify"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.tocify.js"]},{"type":"character","attributes":{},"value":["jquery.tocify.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["navigation"]},{"type":"character","attributes":{},"value":["1.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/navigation-1.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tabsets.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["highlightjs"]},{"type":"character","attributes":{},"value":["9.12.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/highlightjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["highlight.js"]},{"type":"character","attributes":{},"value":["default.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["kePrint"]},{"type":"character","attributes":{},"value":["0.0.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["kePrint-0.0.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["kePrint.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["kableExtra"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.0.1"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["kePrint"]},{"type":"character","attributes":{},"value":["0.0.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["kePrint-0.0.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["kePrint.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["kableExtra"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.0.1"]}]}]}
</script>
<!--/html_preserve-->
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="execution_dependencies">
{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages","version"]},"class":{"type":"character","attributes":{},"value":["data.frame"]},"row.names":{"type":"integer","attributes":{},"value":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137]}},"value":[{"type":"character","attributes":{},"value":["acepack","AnnotationDbi","assertthat","backports","base","base64enc","bindr","bindrcpp","Biobase","BiocGenerics","bit","bit64","blob","caret","checkmate","class","cluster","codetools","colorspace","compiler","cowplot","crayon","data.table","datasets","DBI","DEoptimR","digest","doParallel","doSNOW","dplyr","dynamicTreeCut","evaluate","fastcluster","fit.models","foreach","foreign","Formula","generics","ggplot2","ggpubr","ggrepel","ggsci","glue","GO.db","gower","graphics","grDevices","grid","gridExtra","gtable","highr","Hmisc","hms","htmlTable","htmltools","htmlwidgets","httpuv","httr","impute","ipred","IRanges","iterators","itertools","jsonlite","kableExtra","knitr","labeling","later","lattice","latticeExtra","lava","lazyeval","limma","lubridate","magrittr","MASS","Matrix","matrixStats","memoise","methods","mime","ModelMetrics","munsell","mvtnorm","nlme","nnet","pacman","parallel","pcaPP","pillar","pkgconfig","plotly","plyr","preprocessCore","prodlim","promises","purrr","R6","randomForest","RColorBrewer","Rcpp","readr","recipes","reshape2","rlang","rmarkdown","robust","robustbase","rpart","rrcov","RSQLite","rstudioapi","rvest","S4Vectors","scales","shiny","snow","splines","stats","stats4","stringi","stringr","survival","tibble","tidyr","tidyselect","timeDate","tools","utils","viridisLite","webshot","WGCNA","withr","xfun","xml2","xtable","yaml"]},{"type":"character","attributes":{},"value":["1.4.1","1.44.0","0.2.0","1.1.3","3.5.1","0.1-3","0.1.1","0.2.2","2.42.0","0.28.0","1.1-14","0.9-7","1.1.1","6.0-81","1.9.1","7.3-15","2.0.7-1","0.2-16","1.4-0","3.5.1","0.9.4","1.3.4","1.12.0","3.5.1","1.0.0","1.0-8","0.6.18","1.0.14","1.0.16","0.7.8","1.63-1","0.12","1.1.25","0.5-14","1.4.4","0.8-71","1.2-3","0.0.2","3.1.0","0.2","0.8.0","2.9","1.3.0","3.7.0","0.1.2","3.5.1","3.5.1","3.5.1","2.3","0.2.0","0.7","4.2-0","0.4.2","1.13.1","0.3.6","1.3","1.4.5.1","1.4.0","1.56.0","0.9-8","2.16.0","1.0.10","0.1-3","1.6","1.0.1","1.21","0.3","0.7.5","0.20-38","0.6-28","1.6.5","0.2.1","3.38.3","1.7.4","1.5","7.3-51.1","1.2-15","0.54.0","1.1.0","3.5.1","0.6","1.2.2","0.5.0","1.0-8","3.1-137","7.3-12","0.5.0","3.5.1","1.9-73","1.3.1","2.0.2","4.8.0","1.8.4","1.44.0","2018.04.18","1.0.1","0.3.0","2.3.0","4.6-14","1.1-2","1.0.0","1.3.1","0.1.4","1.4.3","0.3.1","1.11","0.4-18","0.93-3","4.1-13","1.4-7","2.1.1","0.9.0","0.3.2","0.20.1","1.0.0","1.2.0","0.4-3","3.5.1","3.5.1","3.5.1","1.2.4","1.3.1","2.43-3","2.0.1","0.8.2","0.2.5","3043.102","3.5.1","3.5.1","0.3.0","0.5.1","1.66","2.1.2","0.4","1.2.0","1.8-3","2.2.0"]}]}]}
</script>
<!--/html_preserve-->
</div>

<script type="text/javascript" src="../libs/fluidbox/jquery.fluidbox.min.js"></script>
<script type="text/javascript" src="../include/js/enable-fluidbox.js"></script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
