<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Aging and Disease Interaction</title>

<link href='../libs/fluidbox/css/fluidbox.css' rel='stylesheet' type='text/css'>
<link href='https://fonts.googleapis.com/css?family=Source+Code+Pro' rel='stylesheet' type='text/css'>
<link href='../include/css/style.css' rel='stylesheet' type='text/css'>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("section-TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#section-TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#section-TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#section-TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="section-TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <a class="navbar-brand" href="../IndexPage/">
        <img src="../include/gene.png" height="28" width="28">
      </a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="../IndexPage/">Home</a></li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Appendix<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Appendix_PDvsControls_Padlock/>PD vs Controls (padlock)</a></li><li><a href=../Appendix_PDvsControls_RNAseq/>PD vs Controls (RNA-seq)</a></li><li><a href=../Appendix_AgeAcceleration_Padlock/>Age*Diagnosis (padlock)</a></li><li><a href=../Appendix_AgeAcceleration_RNAseq/>Age*Diagnosis (RNAseq)</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Mice<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Mice_CecalPatch_Padlock/>Cecal Patch (padlock)</a></li><li><a href=../Mice_DSS_Padlock/>DSS (padlock)</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Brain<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Brain_PFC_Padlock_CGonly/>PFC II (padlock) CG only</a></li><li><a href=../Brain_PFCRep_Padlock_withGLU/>PFC (large) + GLU (padlock)</a></li><li><a href=../Brain_PFCRep_Padlock_withGLU_Braak/>PFC (large) + GLU Stage-wise (padlock)</a></li><li><a href=../Brain_OFB_Padlock_CGonly/>OFB (padlock) CG only</a></li><li><a href=../Brain_AgeAcceleration_Padlock/>Age*Diagnosis in PFC (large) (padlock)</a></li><li><a href=../Brain_PFC_hMeDIPseq/>PFC (hMeDIP)</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Discover<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Discover_Appendix_Meth2Expression/>Appendix Methylation vs Expression</a></li><li><a href=../Discover_OverlapAppendixMice/>Overlap Appendix & Mice</a></li><li><a href=../Discover_OverlapAppendixBrain/>Overlap Appendix & Brain</a></li><li><a href=../Discover_GenomicEnrichment/>Genomic enrichment</a></li><li><a href=../Discover_Aging/>Aging and Disease Interaction</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Figures<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Figure1/>Figure 1</a></li><li><a href=../Figure2/>Figure 2</a></li><li><a href=../Figure3/>Figure 3</a></li><li><a href=../Figure4/>Figure 4</a></li><li><a href=../Figure5/>Figure 5</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Results<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Results_Appendix/>Appendix</a></li><li><a href=../Results_Brain/>Brain</a></li><li><a href=../Results_Proteomics/>Proteomics</a></li><li><a href=../Results_Aging/>Aging</a></li><li><a href=../Results_Mice/>Mice</a></li><li><a href=../Results_etc/>etc</a></li>
          </ul>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="section-header">



<h1 class="title toc-ignore">Aging and Disease Interaction</h1>

</div>


<!-- # (c)  Juozas Gordevi<c4><8d>ius -->
<div id="section-enrichment-of-pathways" class="section level1">
<h1>Enrichment of pathways</h1>
<div id="section-in-appendix" class="section level2">
<h2>In Appendix</h2>
<pre><code>##     estimate    p.value                      Pathway
## 1: 1.8132895 0.04961678          Selective autophagy
## 2: 1.3505891 0.04867808               Macroautophagy
## 3: 0.7683794 0.04488004                    Lysosomes
## 4: 0.5945298 0.77855878 Chaperone-mediated autophagy
## 5: 1.1576717 0.30251833                    Autophagy</code></pre>
</div>
<div id="section-in-pfc" class="section level2">
<h2>In PFC</h2>
<pre><code>##     estimate     p.value                      Pathway
## 1: 2.1134807 0.009173919          Selective autophagy
## 2: 1.4232660 0.009873693               Macroautophagy
## 3: 0.9318281 0.564812621                    Lysosomes
## 4: 1.1265234 0.635931841 Chaperone-mediated autophagy
## 5: 1.3474460 0.017716298                    Autophagy</code></pre>
</div>
</div>
<div id="section-aging-hallmarks" class="section level1">
<h1>Aging hallmarks</h1>
<pre><code>##       Gene       Score      EPFC      EAPX      Width     Start       End
##  1: RAB33B 0.001757565 1.2291205 1.3766712 0.31922660 0.0000000 0.3192266
##  2:  AP3M2 0.003444828 1.0803242 2.0926480 0.16287072 0.3192266 0.4820973
##  3:  ADRB2 0.005694512 1.0171198 0.9565027 0.09852673 0.4820973 0.5806241
##  4:  AP3S2 0.006569389 1.3040722 0.8832044 0.08540545 0.5806241 0.6660295
##  5:    EN1 0.007897327 0.0000000 5.3616813 0.07104450 0.6660295 0.7370740
##  6:   TBK1 0.007897327 2.4075017 0.0000000 0.07104450 0.7370740 0.8081185
##  7:  PRKCD 0.009029980 0.7910276 1.1181161 0.06213321 0.8081185 0.8702517
##  8:  GNA11 0.009568967 1.3198072 0.8282669 0.05863346 0.8702517 0.9288852
##  9:   GLMP 0.015779031 0.0000000 2.4886084 0.03555742 0.9288852 0.9644426
## 10: GNPTAB 0.015779031 2.2151620 0.0000000 0.03555742 0.9644426 1.0000000</code></pre>
</div>
<div id="section-correlation-of-aging-and-disease-effects" class="section level1">
<h1>Correlation of aging and disease effects</h1>
<div id="section-in-appendix-1" class="section level2">
<h2>In appendix</h2>
<p>Using q &lt; 0.02 as significance threshold.</p>
<p>Overlap of aging in controls and in PDs</p>
<pre><code>##        PD
## CTRL     FALSE   TRUE
##   FALSE 176992   2344
##   TRUE    1846     39</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.007764
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  1.127677 2.195816
## sample estimates:
## odds ratio 
##   1.595226</code></pre>
<p>Overlap of healthy aging and differentially aging loci:</p>
<pre><code>##        DiffA
## CTRL     FALSE   TRUE
##   FALSE 177793   1543
##   TRUE    1852     33</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.0002311
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  1.403920 2.906188
## sample estimates:
## odds ratio 
##   2.053143</code></pre>
<p>Overlap of PD aging and differentially aging loci:</p>
<pre><code>##        DiffA
## PD       FALSE   TRUE
##   FALSE 178676    162
##   TRUE     969   1414</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value &lt; 2.2e-16
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  1321.290 2066.078
## sample estimates:
## odds ratio 
##   1609.586</code></pre>
<blockquote>
<p>Most of the loci did not age in healthy but started showing aging effects in PD</p>
</blockquote>
<p>Overlap of disease and aging loci</p>
<pre class="r"><code>merge(
  apxAg[, list(ID, C = C.DiagnosisControl.Age, 
              P=P.DiagnosisControl.Age, 
              Q = p.adjust(P.DiagnosisControl.Age, &quot;fdr&quot;))],
  apxPD[, list(ID, logFC, P.Value, adj.P.Val)],
  by = &quot;ID&quot;) %&gt;%
  .[, table(Q &lt; 0.2, adj.P.Val &lt; 0.2)] %&gt;%
  tee %&gt;% fisher.test()</code></pre>
<pre><code>##        
##          FALSE   TRUE
##   FALSE 173792   4365
##   TRUE    1831     53</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.2948
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  0.8584986 1.5169810
## sample estimates:
## odds ratio 
##   1.152477</code></pre>
<pre class="r"><code>merge(
  apxAg[, list(ID, C = C.DiagnosisPD.LBD.Age, 
              P=P.DiagnosisPD.LBD.Age, 
              Q = p.adjust(P.DiagnosisControl.Age, &quot;fdr&quot;))],
  apxPD[, list(ID, logFC, P.Value, adj.P.Val)],
  by = &quot;ID&quot;) %&gt;%
  .[, table(Q &lt; 0.2, adj.P.Val &lt; 0.2)] %&gt;%
  tee %&gt;% fisher.test()</code></pre>
<pre><code>##        
##          FALSE   TRUE
##   FALSE 173792   4365
##   TRUE    1831     53</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.2948
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  0.8584986 1.5169810
## sample estimates:
## odds ratio 
##   1.152477</code></pre>
<pre class="r"><code>merge(
  apxAg[, list(ID, C = `C.DiagnosisControl.Age - DiagnosisPD.LBD.Age`, 
              P=`P.DiagnosisControl.Age - DiagnosisPD.LBD.Age`, 
              Q = p.adjust(`P.DiagnosisControl.Age - DiagnosisPD.LBD.Age`, &quot;fdr&quot;))],
  apxPD[, list(ID, logFC, P.Value, adj.P.Val)],
  by = &quot;ID&quot;) %&gt;%
  .[, table(Q &lt; 0.2, adj.P.Val &lt; 0.2)] %&gt;%
  tee %&gt;% fisher.test()</code></pre>
<pre><code>##        
##          FALSE   TRUE
##   FALSE 174111   4360
##   TRUE    1512     58</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.003024
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  1.155492 1.995271
## sample estimates:
## odds ratio 
##   1.531815</code></pre>
<p>Disease effect size correlation with aging effect size</p>
<pre class="r"><code>merge(
  apxAg[, list(ID, C = C.DiagnosisControl.Age, 
              P=P.DiagnosisControl.Age, 
              Q = p.adjust(P.DiagnosisControl.Age, &quot;fdr&quot;))],
  apxPD[, list(ID, logFC, P.Value, adj.P.Val)],
  by = &quot;ID&quot;) %&gt;%
  # .[P &lt; 0.05 &amp; P.Value &lt; 0.05, ] %&gt;%
  .[, cor.test(abs(C), abs(logFC), use=&#39;p&#39;)]</code></pre>
<pre><code>## 
##  Pearson&#39;s product-moment correlation
## 
## data:  abs(C) and abs(logFC)
## t = 162.38, df = 180040, p-value &lt; 2.2e-16
## alternative hypothesis: true correlation is not equal to 0
## 95 percent confidence interval:
##  0.3533808 0.3614390
## sample estimates:
##       cor 
## 0.3574166</code></pre>
<pre class="r"><code>merge(
  apxAg[, list(ID, C = C.DiagnosisPD.LBD.Age, 
              P=P.DiagnosisPD.LBD.Age, 
              Q = p.adjust(P.DiagnosisControl.Age, &quot;fdr&quot;))],
  apxPD[, list(ID, logFC, P.Value, adj.P.Val)],
  by = &quot;ID&quot;) %&gt;%
  # .[P &lt; 0.05 &amp; P.Value &lt; 0.05, ] %&gt;%
  .[, cor.test(abs(C), abs(logFC), use=&#39;p&#39;)]</code></pre>
<pre><code>## 
##  Pearson&#39;s product-moment correlation
## 
## data:  abs(C) and abs(logFC)
## t = 173.11, df = 180040, p-value &lt; 2.2e-16
## alternative hypothesis: true correlation is not equal to 0
## 95 percent confidence interval:
##  0.3737794 0.3816995
## sample estimates:
##       cor 
## 0.3777463</code></pre>
<pre class="r"><code>merge(
  apxAg[, list(ID, C = `C.DiagnosisControl.Age - DiagnosisPD.LBD.Age`, 
              P=`P.DiagnosisControl.Age - DiagnosisPD.LBD.Age`, 
              Q = p.adjust(`P.DiagnosisControl.Age - DiagnosisPD.LBD.Age`, &quot;fdr&quot;))],
  apxPD[, list(ID, logFC, P.Value, adj.P.Val)],
  by = &quot;ID&quot;) %&gt;%
  # .[P &lt; 0.05 &amp; P.Value &lt; 0.05, ] %&gt;%
  .[, cor.test(abs(C), abs(logFC), use=&#39;p&#39;)]</code></pre>
<pre><code>## 
##  Pearson&#39;s product-moment correlation
## 
## data:  abs(C) and abs(logFC)
## t = 173.32, df = 180040, p-value &lt; 2.2e-16
## alternative hypothesis: true correlation is not equal to 0
## 95 percent confidence interval:
##  0.3741697 0.3820871
## sample estimates:
##       cor 
## 0.3781353</code></pre>
</div>
<div id="section-in-pfc-brain" class="section level2">
<h2>In PFC brain</h2>
<p>Using q &lt; 0.02 as significance threshold.</p>
<p>Overlap of aging in controls and in PDs</p>
<pre><code>##        PD
## CTRL     FALSE   TRUE
##   FALSE 824181   1265
##   TRUE    1944      8</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.01181
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  1.153362 5.309810
## sample estimates:
## odds ratio 
##   2.681124</code></pre>
<p>Overlap of healthy aging and differentially aging loci:</p>
<pre><code>##        DiffA
## CTRL     FALSE   TRUE
##   FALSE 824913    533
##   TRUE    1849    103</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value &lt; 2.2e-16
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##   68.81245 107.49014
## sample estimates:
## odds ratio 
##    86.2173</code></pre>
<p>Overlap of PD aging and differentially aging loci:</p>
<pre><code>##        DiffA
## PD       FALSE   TRUE
##   FALSE 825758    367
##   TRUE    1004    269</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value &lt; 2.2e-16
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  506.5601 716.9624
## sample estimates:
## odds ratio 
##   602.7873</code></pre>
<blockquote>
<p>Most of the loci did not age in healthy but started showing aging effects in PD</p>
</blockquote>
<p>Overlap of disease and aging loci</p>
<pre class="r"><code>merge(
  pfcAg[, list(ID, C = C.GroupCTRL.Age, 
              P=P.GroupCTRL.Age, 
              Q = p.adjust(P.GroupCTRL.Age, &quot;fdr&quot;))],
  pfcPD[, list(ID, logFC, P.Value, adj.P.Val)],
  by = &quot;ID&quot;) %&gt;%
  .[, table(P &lt; 0.05, adj.P.Val &lt; 0.05)] %&gt;%
  tee %&gt;% fisher.test()</code></pre>
<pre><code>##        
##          FALSE   TRUE
##   FALSE 773237     62
##   TRUE   54091      8</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.1378
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  0.7626883 3.8656513
## sample estimates:
## odds ratio 
##   1.844463</code></pre>
<pre class="r"><code>merge(
  pfcAg[, list(ID, C = C.GroupPD.Age, 
              P=P.GroupPD.Age, 
              Q = p.adjust(P.GroupCTRL.Age, &quot;fdr&quot;))],
  pfcPD[, list(ID, logFC, P.Value, adj.P.Val)],
  by = &quot;ID&quot;) %&gt;%
  .[, table(P &lt; 0.05, adj.P.Val &lt; 0.05)] %&gt;%
  tee %&gt;% fisher.test()</code></pre>
<pre><code>##        
##          FALSE   TRUE
##   FALSE 773567     64
##   TRUE   53761      6</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.4616
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  0.4772995 3.1008821
## sample estimates:
## odds ratio 
##   1.348959</code></pre>
<pre class="r"><code>merge(
  pfcAg[, list(ID, C = `C.GroupCTRL.Age - GroupPD.Age`, 
              P=`P.GroupCTRL.Age - GroupPD.Age`, 
              Q = p.adjust(`P.GroupCTRL.Age - GroupPD.Age`, &quot;fdr&quot;))],
  pfcPD[, list(ID, logFC, P.Value, adj.P.Val)],
  by = &quot;ID&quot;) %&gt;%
  .[, table(P &lt; 0.05, adj.P.Val &lt; 0.05)] %&gt;%
  tee %&gt;% fisher.test()</code></pre>
<pre><code>##        
##          FALSE   TRUE
##   FALSE 773725     62
##   TRUE   53603      8</code></pre>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  .
## p-value = 0.1364
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  0.7701127 3.9028996
## sample estimates:
## odds ratio 
##   1.862399</code></pre>
<p>Disease effect size correlation with aging effect size</p>
<pre class="r"><code>merge(
  pfcAg[, list(ID, C = C.GroupCTRL.Age, 
              P=P.GroupCTRL.Age, 
              Q = p.adjust(P.GroupCTRL.Age, &quot;fdr&quot;))],
  pfcPD[, list(ID, logFC, P.Value, adj.P.Val)],
  by = &quot;ID&quot;) %&gt;%
  #.[P &lt; 0.05 &amp; P.Value &lt; 0.05, ] %&gt;%
  .[, cor.test( abs(C), abs(logFC), use=&#39;s&#39;)]</code></pre>
<pre><code>## 
##  Pearson&#39;s product-moment correlation
## 
## data:  abs(C) and abs(logFC)
## t = 293.55, df = 827400, p-value &lt; 2.2e-16
## alternative hypothesis: true correlation is not equal to 0
## 95 percent confidence interval:
##  0.3051683 0.3090712
## sample estimates:
##      cor 
## 0.307121</code></pre>
<pre class="r"><code>merge(
  pfcAg[, list(ID, C = C.GroupPD.Age, 
              P=P.GroupPD.Age, 
              Q = p.adjust(P.GroupPD.Age, &quot;fdr&quot;))],
  pfcPD[, list(ID, logFC, P.Value, adj.P.Val)],
  by = &quot;ID&quot;) %&gt;%
  # .[P &lt; 0.05 &amp; P.Value &lt; 0.05, ] %&gt;%
  .[, cor.test(abs(C), abs(logFC), use=&#39;p&#39;)]</code></pre>
<pre><code>## 
##  Pearson&#39;s product-moment correlation
## 
## data:  abs(C) and abs(logFC)
## t = 285.82, df = 827400, p-value &lt; 2.2e-16
## alternative hypothesis: true correlation is not equal to 0
## 95 percent confidence interval:
##  0.2978120 0.3017342
## sample estimates:
##       cor 
## 0.2997744</code></pre>
<pre class="r"><code>merge(
  pfcAg[, list(ID, C = `C.GroupCTRL.Age - GroupPD.Age`, 
              P=`P.GroupCTRL.Age - GroupPD.Age`, 
              Q = p.adjust(`P.GroupCTRL.Age - GroupPD.Age`, &quot;fdr&quot;))],
  pfcPD[, list(ID, logFC, P.Value, adj.P.Val)],
  by = &quot;ID&quot;) %&gt;%
  # .[P &lt; 0.05 &amp; P.Value &lt; 0.05, ] %&gt;%
  .[, cor.test(abs(C), abs(logFC), use=&#39;p&#39;)]</code></pre>
<pre><code>## 
##  Pearson&#39;s product-moment correlation
## 
## data:  abs(C) and abs(logFC)
## t = 289.29, df = 827400, p-value &lt; 2.2e-16
## alternative hypothesis: true correlation is not equal to 0
## 95 percent confidence interval:
##  0.3011242 0.3050378
## sample estimates:
##       cor 
## 0.3030823</code></pre>
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="dependencies">
{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["bootstrap"]},{"type":"character","attributes":{},"value":["3.3.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/bootstrap"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["viewport"]}},"value":[{"type":"character","attributes":{},"value":["width=device-width, initial-scale=1"]}]},{"type":"character","attributes":{},"value":["js/bootstrap.min.js","shim/html5shiv.min.js","shim/respond.min.js"]},{"type":"character","attributes":{},"value":["css/bootstrap.min.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jqueryui"]},{"type":"character","attributes":{},"value":["1.11.4"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jqueryui"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery-ui.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tocify"]},{"type":"character","attributes":{},"value":["1.9.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/tocify"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.tocify.js"]},{"type":"character","attributes":{},"value":["jquery.tocify.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["navigation"]},{"type":"character","attributes":{},"value":["1.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/navigation-1.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tabsets.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["highlightjs"]},{"type":"character","attributes":{},"value":["9.12.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/highlightjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["highlight.js"]},{"type":"character","attributes":{},"value":["default.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.11"]}]}]}
</script>
<!--/html_preserve-->
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="execution_dependencies">
{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages","version"]},"class":{"type":"character","attributes":{},"value":["data.frame"]},"row.names":{"type":"integer","attributes":{},"value":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123]}},"value":[{"type":"character","attributes":{},"value":["acepack","AnnotationDbi","assertthat","backports","base","base64enc","bindr","bindrcpp","Biobase","BiocGenerics","bit","bit64","blob","broom","checkmate","cluster","codetools","colorspace","compiler","crayon","data.table","datasets","DBI","DEoptimR","digest","doParallel","doSNOW","dplyr","dynamicTreeCut","evaluate","fastcluster","fit.models","foreach","foreign","Formula","generics","ggplot2","ggpubr","glue","GO.db","graphics","grDevices","grid","gridExtra","gtable","Hmisc","hms","htmlTable","htmltools","htmlwidgets","httpuv","httr","impute","IRanges","iterators","itertools","jsonlite","kableExtra","knitr","later","lattice","latticeExtra","lazyeval","limma","magrittr","MASS","Matrix","matrixStats","memoise","methods","mime","munsell","mvtnorm","nlme","nnet","pacman","parallel","pcaPP","pillar","pkgconfig","plotly","plyr","preprocessCore","promises","purrr","R6","RColorBrewer","Rcpp","readr","rlang","rlecuyer","rmarkdown","robust","robustbase","RobustRankAggreg","rpart","rrcov","RSQLite","rstudioapi","rvest","S4Vectors","scales","shiny","snow","splines","stats","stats4","stringi","stringr","survival","tibble","tidyr","tidyselect","tools","utils","viridisLite","webshot","WGCNA","withr","xfun","xml2","xtable","yaml"]},{"type":"character","attributes":{},"value":["1.4.1","1.44.0","0.2.0","1.1.3","3.5.1","0.1-3","0.1.1","0.2.2","2.42.0","0.28.0","1.1-14","0.9-7","1.1.1","0.5.1","1.9.1","2.0.7-1","0.2-16","1.4-0","3.5.1","1.3.4","1.12.0","3.5.1","1.0.0","1.0-8","0.6.18","1.0.14","1.0.16","0.7.8","1.63-1","0.12","1.1.25","0.5-14","1.4.4","0.8-71","1.2-3","0.0.2","3.1.0","0.2","1.3.0","3.7.0","3.5.1","3.5.1","3.5.1","2.3","0.2.0","4.2-0","0.4.2","1.13.1","0.3.6","1.3","1.4.5.1","1.4.0","1.56.0","2.16.0","1.0.10","0.1-3","1.6","1.0.1","1.21","0.7.5","0.20-38","0.6-28","0.2.1","3.38.3","1.5","7.3-51.1","1.2-15","0.54.0","1.1.0","3.5.1","0.6","0.5.0","1.0-8","3.1-137","7.3-12","0.5.0","3.5.1","1.9-73","1.3.1","2.0.2","4.8.0","1.8.4","1.44.0","1.0.1","0.3.0","2.3.0","1.1-2","1.0.0","1.3.1","0.3.1","0.3-4","1.11","0.4-18","0.93-3","1.1","4.1-13","1.4-7","2.1.1","0.9.0","0.3.2","0.20.1","1.0.0","1.2.0","0.4-3","3.5.1","3.5.1","3.5.1","1.2.4","1.3.1","2.43-3","2.0.1","0.8.2","0.2.5","3.5.1","3.5.1","0.3.0","0.5.1","1.66","2.1.2","0.4","1.2.0","1.8-3","2.2.0"]}]}]}
</script>
<!--/html_preserve-->
</div>
</div>

<script type="text/javascript" src="../libs/fluidbox/jquery.fluidbox.min.js"></script>
<script type="text/javascript" src="../include/js/enable-fluidbox.js"></script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
